src/reformat_libsharpyuv.c

```diff
@@ -9,6 +9,11 @@
 
 avifResult avifImageRGBToYUVLibSharpYUV(avifImage * image, const avifRGBImage * rgb, const avifReformatState * state)
 {
+    // The width, height, and stride parameters of SharpYuvConvertWithOptions()
+    // and SharpYuvConvert() are all of the int type.
+    if (rgb->width > INT_MAX || rgb->height > INT_MAX || rgb->rowBytes > INT_MAX || image->yuvRowBytes[AVIF_CHAN_Y] > INT_MAX) {
+        return AVIF_RESULT_NOT_IMPLEMENTED;
+    }
```

Comment:
Rather than checking this only when we are calling into libyuv or libsharpyuv, I am wondering if we should check this in the public `avifImageRGBToYUV()` and `avifImageYUVToRGB()` functions.

Reply:
We can do both to be safe. I prefer to keep the check close to where the values are used.

Reply:
Yannis: Thank you for your comment. I also just realized that the maximum width and height that AVIF can achieve may be smaller than INT_MAX. We calculated the maximum width and height of AVIF before (assuming max AV1 frame size of 64K x 64K and the maximum HEIF/MIAF image grid), but I can't find that email thread.

On the other hand, the `avifImageRGBToYUV()` and `avifImageYUVToRGB()` functions are really independent of AVIF, so it may be good to let them "be all they can be" and only impose the limits related to YUV<->RGB conversions.

I'd like to think about this a bit more, so I won't merge this pull request right away.